src/ai/backend/manager/scheduler/dispatcher.py

```diff
@@ -619,24 +628,95 @@ async def _schedule_single_node_session(
             # If sess_ctx.agent_id is already set for manual assignment by superadmin,
             # skip assign_agent_for_session().
             agent = sess_ctx.main_kernel.agent_row
-            agent_id: AgentId
+            agent_id: AgentId | None = None
             if agent is not None:
                 agent_id = agent.id
             else:
-                # Let the scheduler check the resource availability and decide the target agent
-                cand_agent = scheduler.assign_agent_for_session(
-                    compatible_candidate_agents, sess_ctx
-                )
-                if cand_agent is None:
-                    raise InstanceNotAvailable(
-                        extra_msg=(
-                            "Could not find a contiguous resource region in any agent "
-                            "big enough to host the session "
-                            f"(id: {sess_ctx.id}, resource group: {sess_ctx.scaling_group_name})"
-                        ),
+                sorted_agents = sorted(compatible_candidate_agents, key=lambda agent: agent.id)
+
+                if scheduler.sgroup_opts.roundrobin:
+                    rr_state_str: str | None = await sched_ctx.registry.shared_config.get_raw(
+                        "roundrobin_state"
```

Comment:
In case of defining customized type, we can define new method in `shared_config`, such as `get_round_robin_state()` and then cast(or parse) the string into `dict` or customized type.